app/waf/src/controllers/application.controller.ts

```diff
@@ -278,8 +278,11 @@ export class ApplicationController extends BaseController {
       tenantId: tenantId,
     });
 
+    let mgmt = adc.management!;
+    let asgManager = await ASGManager.instanlize();
+
     let operation = patchOP.Remove;
     let req = new AS3PatchReqeust(adc, application, operation);
-    return await this.as3Service.deploy(AS3_HOST, AS3_PORT, req);
+    await asgManager.deploy(mgmt.ipAddress, mgmt.tcpPort, req);
```

Comment:
This call to deploy may need to be surrounded with try..catch..422 or 4xx, or it may response 500 error. Same case for `deployById` maybe.

Reply:
for deployById, it seems that's because it needs to consider when the partition is not there at all, and will try to use a full-deploy instead to try again. does not seem to need that part here.

Reply:
Do you mean this part of code will be replaced with some else? So no need to consider it for now?

Reply:
what i meant is that in the deployById(), they added the try/catch because they need to try "deploy-method" when "patch-method" fails(in case the partition is not found on bigip), we don't seem to need this here.

Reply:
Just f2f talk, and got the core, in the future, we may need to refine the exception catching in several places of the code.